utils/land-patterns.stanza

```diff
@@ -71,264 +204,297 @@ public defn soldermask-defined (p:Pad):
+
+; Create a circular non-plated hole landpattern with given drill radius and mask radius
+public defn non-plated-hole-landpattern (drill-r:Double, mask-r:Double) :
+  non-plated-hole-landpattern(Circle(drill-r), Circle(mask-r))
+
+; Create a non-plated hole landpattern with a given drill radius.
+public defn non-plated-hole-landpattern (drill-r:Double) :
+  non-plated-hole-landpattern(drill-r, drill-r)
+
+;==============================================================================
+;========================== Land Pattern Generators ===========================
+;==============================================================================
+
+;==============================================================================
+;========================== Pose Arrangement Helpers ==========================
+;==============================================================================
+; PinPadMapping is a not-yet-placed pad, with assigned name, pad, and pose
+public pcb-struct ocdb/land-patterns/PinPadMapping:
+  pad-name:  Ref,  ; the name of the pad
+  pad-shape: JITXDef, ; the corresponding pcb-pad
+  pose:      Pose,    ; the pose of the pad
+
+; A generic landpattern created from a sequence of PinPadMappings
+public defn make-generic-landpattern (pads:Seqable<PinPadMapping>):
+  inside pcb-landpattern:
+    for p in pads do:
+      pad (pad-name(p)): (pad-shape(p) as Pad) at pose(p)
+
+; Create a generic landpattern with a uniform pad
+public defn make-generic-landpattern (pad-names:Seqable<Ref>,
+                                      poses:Seqable<Pose>,
+                                      pad-shape:JITXDef):
+  val pads = 
+    for (pn in pad-names, pp in poses) seq:
+      PinPadMapping(pn, pad-shape, pp) 
+  make-generic-landpattern(to-tuple(pads))
+
+; Naming for a 2-column arrangment of pins 
+public defn soic-naming-convention (row:Int, column:Int, num-pins:Int) -> Ref:
+  if (column < 0) or (column > 1): 
+    fatal("soic naming convention supports a maximum of 2 columns")
+  switch(column):
+    0: IndexRef(Ref("p"), row + 1)
+    1: IndexRef(Ref("p"), num-pins - row)
+  
+; Create a two-column landpattern
+public defn make-two-column-landpattern (count: Int,    ; number of pads in the column
+                                         pitch: Double, ; pitch of the column
+                                         naming-convention: (Int, Int) -> Ref, ; naming convention
+                                         pad-convention: (Int, Int) -> JITXDef,
+                                         spacing:Double):
+  val mapping = to-tuple $
+    for col in 0 to 2 seq-cat:
+      val column = pose-column(loc(to-double(col) * spacing, 0.0), count, pitch)
+      for (pose in column, row in 0 to false) seq:
+        PinPadMapping(naming-convention(row, col), pad-convention(row, col), pose)
+  make-generic-landpattern(mapping)
+
+public defn make-grid-landpattern (
+  num-rows:Int,
```

Comment:
Please use idiomatic indentation for argument lists.

Reply:
Fixed